pmd-core/src/main/java/net/sourceforge/pmd/util/log/MessageReporter.java

```diff
@@ -95,4 +96,13 @@ default void errorEx(String message, Object[] formatArgs, Throwable error) {
      */
     int numErrors();
 
+
+    /**
+     * Returns a dummy instance that does not output anything, but still
+     * counts errors.
+     */
+    static MessageReporter noop() {
+        return new NoopReporter();
```

Comment:
this can probably be a singleton, no need to create new instances each time

Reply:
No it can't, because as documented, it counts errors, so it has some internal state. It is noop in the sense that it does not render messages, but the error count is used to abort analysis in some places, and not counting errors would break this feature.

Reply:
Oh, I see… It is a little counterintuitive though, and unlike other `noop` we expose (ie: `FileAnalysisListener.noop`)… we should probably just rename it to something like `ErrorCountingReporter` or `VoidReporter` along this method, WDYT?

Reply:
yeah, sounds good.

Reply:
Mabye something like `QuietReporter` or `SilentReporter`?

Reply:
I picked QuietReporter